File: src/main/java/todayquest/quest/dto/DetailQuestResponseDto.java
package todayquest.quest.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import todayquest.quest.entity.DetailQuest;
import todayquest.quest.entity.DetailQuestState;
import todayquest.quest.entity.DetailQuestType;

@Data
@NoArgsConstructor
@AllArgsConstructor @Builder
public class DetailQuestResponseDto {
    private String title;
    private Short count;
    private DetailQuestType type;
    private DetailQuestState state;

    public static DetailQuestResponseDto createDto(DetailQuest dq) {
        return DetailQuestResponseDto.builder()
                .title(dq.getTitle())
                .count(dq.getTargetCount())
                .type(dq.getType())
                .state(dq.getState())
                .build();
    }
}

File: src/main/java/todayquest/quest/dto/DetailQuestResponseDto.kt
package todayquest.quest.dto

import todayquest.quest.entity.DetailQuest
import todayquest.quest.entity.DetailQuestState
import todayquest.quest.entity.DetailQuestType

data class DetailQuestResponseDto(
    val id: Long? = null,
    val title: String? = null,
    val targetCount: Short? = null,
    val count: Short? = null,
    val type: DetailQuestType? = null,
    val state: DetailQuestState? = null,
) {

    companion object {
        fun createDto(dq: DetailQuest): DetailQuestResponseDto {
            return DetailQuestResponseDto(
                id = dq.id,
                title = dq.title,
                targetCount = dq.targetCount,
                count = dq.count,
                type = dq.type,
                state = dq.state
            )
        }
    }
}